### PR TITLE
[Feature] Enables a public Trust Chain endpoint

### PR DESCRIPTION
**Pull request by justynspooner**
This PR provides a new env variable that allows the trustchain API to be publically exposed so anyone could query it for a strucutred trustchain payload.

Add `PUBLIC_TRUST_CHAIN_ACCESS=TRUE` to the environment to enable

### CODE REVIEW

File: src/handler/policies/trustChainsHandler.ts [hunk 1]
```diff
@@ -3,14 +3,29 @@ import Response from 'src/response'
...
 
 async function TrustChainsHandler(
 	req: GuardianMiddlewareRequest,
 	res: NextApiResponse
 ) {
 	const { policyId } = req.query
 	const { guardian } = req.context
-	const { accessToken } = req
+
+	let { accessToken } = req
```
Comment (mattsmithies) [inline src/handler/policies/trustChainsHandler.ts:16]: In the house of DOVU we don't use lets, transform this into a separate method or inline function

File: src/config/index.ts [hunk 2]
```diff
@@ -56,6 +58,7 @@ const config: Config = {
 	guardianApiUrl: GUARDIAN_API_URL,
 	registryUsername: STANDARD_REGISTRY_USERNAME,
 	registryPassword: STANDARD_REGISTRY_PASSWORD,
+	publicTrustChainAccess: true, // booleanValue(PUBLIC_TRUST_CHAIN_ACCESS),
```
Comment (justynspooner) [inline src/config/index.ts:61]:
```suggestion
	publicTrustChainAccess: booleanValue(PUBLIC_TRUST_CHAIN_ACCESS),
```

File: pages/api/policies/[policyId]/trustchains/index.ts [hunk 3]
```diff
@@ -7,12 +7,16 @@ import ensureRole from 'src/middleware/ensureRole'
...
+const emptyHandler = (handler) => (req, res) => handler(req, res)
 
 export default prepare(
 	exceptionFilter,
 	onlyGet,
-	withHmac,
 	useGuardianContext,
-	withAuthentication,
-	ensureRole(Role.STANDARD_REGISTRY)
+	isProtected ? withHmac : emptyHandler,
```
Comment (mattsmithies) [inline pages/api/policies/[policyId]/trustchains/index.ts:19]: This is nonsense, split into a separate method above prepare.

```js
const generateMiddleware = () => {

  const base = [middlewares]

  if (!config.publicTrustChainAccess) {
    return [
       ...base
       (extras)
    ]
  }
  
  return base

  }

export default prepare(
  ...generateMiddleware()
)
```

File: pages/api/policies/[policyId]/trustchains/index.ts [hunk 4]
```diff
@@ -7,12 +7,16 @@ import ensureRole from 'src/middleware/ensureRole'
 import { Role } from 'src/config/guardianTags'
 import exceptionFilter from 'src/middleware/exceptionFilter'
 import onlyGet from 'src/middleware/onlyGet'
+import config from 'src/config'
+
+const isProtected = !config.publicTrustChainAccess
+const emptyHandler = (handler) => (req, res) => handler(req, res)
 
 export default prepare(
 	exceptionFilter,
```
Comment (mattsmithies) [inline pages/api/policies/[policyId]/trustchains/index.ts:16]: Remember to wrap exception filter soon

**Review by mattsmithies — CHANGES_REQUESTED** [review]
Gimme a call if you want to go through the changes

File: pages/api/policies/[policyId]/trustchains/index.ts (hunk 4 above)
Comment (justynspooner) [reply pages/api/policies/[policyId]/trustchains/index.ts]: Done

File: pages/api/policies/[policyId]/trustchains/index.ts (hunk 3 above)
Comment (justynspooner) [reply pages/api/policies/[policyId]/trustchains/index.ts]: Thank you ser

File: src/handler/policies/trustChainsHandler.ts (hunk 1 above)
Comment (justynspooner) [reply src/handler/policies/trustChainsHandler.ts]: Let us not see let again. It has been squashed.

**Comment by justynspooner** [comment]
@mattsmithies I've addressed those comments. Should be good for another review now.

**Review by mattsmithies — APPROVED** [review]

**Pull request merged**